CMakeLists.txt

```diff
@@ -46,6 +47,14 @@ if(BUILD_TESTING)
 
   ament_lint_auto_find_test_dependencies()
 
+  ament_add_gtest(test_asserts_ndebug test/test_asserts.cpp)
+  target_link_libraries(test_asserts_ndebug ${PROJECT_NAME})
```

Comment:
I think you still might want to check if `test_asserts_ndebug` is a target before using it, as I do not think `ament_add_gtest` guarantees that it creates a target, in the case that google test is not found:

https://github.com/ament/ament_cmake/blob/83e1b8147e2eda7d805e45c163aea60230a4d160/ament_cmake_gtest/cmake/ament_add_gtest.cmake#L21-L22